src/utils/method-helpers.js

```diff
@@ -28,3 +29,6 @@ const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
 
 export const delay = (delayed, ms) =>
   Promise.all([delayed, sleep(ms)]).then(([data]) => data);
+
+export const translate = (key, lang, interpolation = null) =>
```

Comment:
@housseindjirdeh well, as I see: we determine the language of the device, if such is supported by the application, we use it. Further, if the user has changed the language in the application settings (we save it via AsyncStorage), then we use it (ie the language in the application settings has priority over the device language). 

If this language is not supported by the application, then English is set by default. When the user opens  the application again, we look whether there is a selected language in the application settings (via AsyncStorage), if there is one we use it.

> Just calling the translate function should suffice I think.

This is what I had in mind in order to discard this parameter.